integration/integration.go

```diff
@@ -0,0 +1,586 @@
+
+		if root.HorizonSequence < 3 ||
+			int(root.HorizonSequence) != int(root.IngestSequence) {
+			continue
+		}
+
+		if uint32(root.CurrentProtocolVersion) != 0 {
+			i.t.Logf("Horizon protocol version upgraded to %d",
+				root.CurrentProtocolVersion)
+			return
+		}
+	}
+
+	i.t.Fatal("Horizon not ingesting...")
+}
+
+func (i *Test) waitForFriendbot() {
+	for t := 60; t >= 0; t -= 1 {
+		time.Sleep(time.Second)
+
+		i.t.Log("Waiting for friendbot...")
+		url := fmt.Sprintf("http://%s:8000/friendbot", dockerHost)
+		resp, err := http.Get(url)
+		if err != nil {
+			continue
+		}
+
+		if resp.StatusCode == http.StatusBadGateway {
+			continue
+		}
+
+		return
+	}
+
+	i.t.Fatal("Friendbot not working...")
+}
+
+func (it *Test) waitForStarbridge(count int) {
+	g := new(errgroup.Group)
+
+	for i := 0; i < count; i++ {
+		i := i
+		g.Go(func() error {
+			for t := 60; t >= 0; t -= 1 {
+				time.Sleep(time.Second)
+
+				port := 9000 + i
+				it.t.Logf("Waiting for Starbridge at port %d...", port)
+				url := fmt.Sprintf("http://localhost:%d", port)
+				_, err := http.Get(url)
+				if err != nil {
+					continue
+				}
+
+				return nil
+			}
+
+			return errors.New("Starbridge not responding...")
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		it.t.Fatal(err)
+	}
+}
+
+// HorizonClient returns horizon.Client connected to started Horizon instance.
+func (i *Test) HorizonClient() *horizonclient.Client {
+	return i.horizonClient
+}
+
+// Client returns http.Client connected to started Starbridge instance.
+func (i *Test) Client() *http.Client {
+	return i.client
+}
+
+// StopHorizon shuts down the running Horizon process
+// func (i *Test) StopStarbridge() {
+// 	i.app.Close()
+
+// 	// Wait for Horizon to shut down completely.
+// 	<-i.appStopped
+
+// 	i.app = nil
+// }
+
+// Master returns a keypair of the network masterKey account.
+func (i *Test) Master() *keypair.Full {
+	if i.masterKey != nil {
+		return i.masterKey
+	}
+	return keypair.Master(i.passPhrase).(*keypair.Full)
+}
+
+func (i *Test) MasterAccount() txnbuild.Account {
+	account := i.MasterAccountDetails()
+	return &account
+}
+
+func (i *Test) MasterAccountDetails() proto.Account {
+	return i.MustGetAccount(i.Master())
+}
+
+func (i *Test) CurrentTest() *testing.T {
+	return i.t
+}
+
+/* Utility functions for easier test case creation. */
+
+// Creates new accounts via friendbot.
+//
+// Returns: The slice of created keypairs and account objects.
+//
+// Note: panics on any errors, since we assume that tests cannot proceed without
+// this method succeeding.
+func (i *Test) CreateAccounts(count int) ([]*keypair.Full, []txnbuild.Account) {
+	pairs := make([]*keypair.Full, count)
+	accounts := make([]txnbuild.Account, count)
+
+	g := new(errgroup.Group)
+
+	for j := range pairs {
+		j := j
+		g.Go(func() error {
+			pair, _ := keypair.Random()
+			pairs[j] = pair
+
+			url := fmt.Sprintf("http://%s:8000/friendbot?addr=%s", dockerHost, pair.Address())
+			resp, err := http.Get(url)
```

Comment:
FYI you can use the horizonclient.Client.Fund() function, it works with any network, not just testnet.